src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1496,7 +1496,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
 
                 case SymbolKind.Alias:
                     {
-                        var alias = symbol as AliasSymbol;
+                        var alias = (AliasSymbol)symbol;
                         symbol = alias.Target;
```

Comment:
The perspective I've been thinking about this pattern is for Watson bugs.  There are two specific types of bugs that could happen on this code path that interested me:

- NullReferenceException 
- The cast fails

In the `as` pattern both problems end up with the same symptom: NRE.  That makes it harder to detect when looking at a crash dump.  In the cast pattern the two errors would be pretty distinct in a Watson bug.